src/sql/workbench/browser/modelComponents/listView.component.ts

```diff
@@ -0,0 +1,182 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { ChangeDetectorRef, Component, ElementRef, forwardRef, Inject, Input, OnDestroy, QueryList, ViewChildren } from '@angular/core';
+import * as azdata from 'azdata';
+import { ComponentBase } from 'sql/workbench/browser/modelComponents/componentBase';
+import * as DOM from 'vs/base/browser/dom';
+import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { KeyCode } from 'vs/base/common/keyCodes';
+
+import 'vs/css!./media/listView';
+
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/platform/dashboard/browser/interfaces';
+import { deepClone } from 'vs/base/common/objects';
+
+@Component({
+	templateUrl: decodeURI(require.toUrl('./listView.component.html'))
+})
+
+export default class ListViewComponent extends ComponentBase<azdata.ListViewComponentProperties> implements IComponent, OnDestroy {
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+	@ViewChildren('optionDiv') optionElements: QueryList<ElementRef>;
+
+	private focusedOptionId: string | undefined;
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) changeRef: ChangeDetectorRef,
+		@Inject(forwardRef(() => ElementRef)) el: ElementRef,
+	) {
+		super(changeRef, el);
+	}
+
+	ngOnInit(): void {
+		this.baseInit();
+	}
+
+	setLayout(layout: any): void {
+		this.layout();
+	}
+
+	ngOnDestroy(): void {
+		this.baseDestroy();
+	}
+
+	onKeyDown(event: KeyboardEvent): void {
+		if (!this.enabled || this.options.length === 0) {
+			return;
+		}
+		let e = new StandardKeyboardEvent(event);
+		if (e.keyCode === KeyCode.Enter || e.keyCode === KeyCode.Space) {
+			if (this.focusedOptionId) {
+				this.selectOption(this.focusedOptionId);
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+		else if (e.keyCode === KeyCode.LeftArrow || e.keyCode === KeyCode.UpArrow) {
+			if (this.focusedOptionId) {
+				this.focusOption(this.findPreviousOption(this.focusedOptionId));
+			}
+			DOM.EventHelper.stop(e, true);
+		} else if (e.keyCode === KeyCode.RightArrow || e.keyCode === KeyCode.DownArrow) {
+			if (this.focusedOptionId) {
+				this.focusOption(this.findNextOption(this.focusedOptionId));
+			}
+			DOM.EventHelper.stop(e, true);
+		}
+	}
+
+	private getOptionById(optionId: string): azdata.ListViewOption {
+		const filteredOptions = this.options.filter(o => { return o.id === optionId; });
+		if (filteredOptions.length === 1) {
+			return filteredOptions[0];
+		} else {
+			throw new Error(`There should be one and only one matching option for the giving option id, actual number: ${filteredOptions.length}, option id: ${optionId}.`);
```

Comment:
`for the given option id '${optionId}'. Actual number...`

Although this seems like a very odd place to be doing this check - if you think this is necessary you should be doing it where the options are set instead.